pdks/Gf180/gf180/test_site_sims.py

```diff
@@ -0,0 +1,303 @@
+                d=vdd, g=vdd, s=VSS, b=VSS
+            )
+            pfet_03v3_dss = g.PFET_3p3V_DSS(p(w=0.220 * µ, l=0.280 * µ))(
+                d=vdd, g=vdd, s=VSS, b=VSS
+            )
+            # nfet_06v0_dss = g.NFET_6p0V_DSS(p(w = 0.300 * µ,l = 0.500 * µ))(d=vdd, g=vdd, s=VSS, b=VSS)
+            pfet_06v0_dss = g.PFET_6p0V_DSS(p(w=0.300 * µ, l=0.500 * µ))(
+                d=vdd, g=vdd, s=VSS, b=VSS
+            )
+
+            nfet_06v0_nvt = g.NFET_6p0V_NAT(p(w=0.800 * µ, l=1.800 * µ))(
+                d=vdd, g=vdd, s=VSS, b=VSS
+            )
+
+        # Simulation Controls
+        op = h.sim.Op()
+        i1 = gf180.install.include_design()
+        i2 = gf180.install.include_mos(Corner.TYP)
+
+    opts = vsp.SimOptions(
+        simulator=vsp.SupportedSimulators.NGSPICE,
+        fmt=vsp.ResultFormat.SIM_DATA,
+        rundir="./scratch",
+    )
+
+    rv = Sim.run(opts)
+    assert isinstance(rv, vsp.sim_data.SimResult)
+
+    op = rv[vsp.sim_data.AnalysisType.OP]
+    assert isinstance(op, vsp.sim_data.OpResult)
+
+
+def test_sim_resistors():
+    @h.sim.sim
+    class Sim:
+        @h.module
+        class Tb:
+
+            VSS = h.Port()
+            vdd = h.Signal()
+            v = h.Vdc(dc=1)(p=vdd, n=VSS)
+
+            p = gf180.GF180ResParams()
+
+            # Three terminal resistors
+            nplus_u = g.NPLUS_U(p)(p=vdd, n=VSS, b=VSS)
+            pplus_u = g.PPLUS_U(p)(p=vdd, n=VSS, b=VSS)
+            nplus_s = g.NPLUS_S(p)(p=vdd, n=VSS, b=VSS)
+            pplus_s = g.PPLUS_S(p)(p=vdd, n=VSS, b=VSS)
+            nwell = g.NWELL(p)(p=vdd, n=VSS, b=VSS)
+            npolyf_u = g.NPOLYF_U(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u = g.PPOLYF_U(p)(p=vdd, n=VSS, b=VSS)
+            npolyf_s = g.NPOLYF_S(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_s = g.PPOLYF_S(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u_1k = g.PPOLYF_U_1K(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u_2k = g.PPOLYF_U_2K(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u_1k_6p0 = g.PPOLYF_U_1K_6P0(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u_2k_6p0 = g.PPOLYF_U_2K_6P0(p)(p=vdd, n=VSS, b=VSS)
+            ppolyf_u_3k = g.PPOLYF_U_3K(p)(p=vdd, n=VSS, b=VSS)
+
+            # Two terminal resistors
+            rm1 = g.RM1(p)(p=vdd, n=VSS)
+            rm2 = g.RM2(p)(p=vdd, n=VSS)
+            rm3 = g.RM3(p)(p=vdd, n=VSS)
+            tm6k = g.TM6K(p)(p=vdd, n=VSS)
+            tm9k = g.TM9K(p)(p=vdd, n=VSS)
+            tm11k = g.TM11K(p)(p=vdd, n=VSS)
+            tm30k = g.TM30K(p)(p=vdd, n=VSS)
+
+        # Simulation Controls
+        op = h.sim.Op()
+
+        d1 = gf180.install.include_design()
+        i1 = gf180.install.include_mos(Corner.TYP)
+        i2 = gf180.install.include_resistors(Corner.TYP)
+
+    opts = vsp.SimOptions(
+        simulator=vsp.SupportedSimulators.NGSPICE,
+        fmt=vsp.ResultFormat.SIM_DATA,
+        rundir="./scratch",
+    )
+
+    rv = Sim.run(opts)
+    assert isinstance(rv, vsp.sim_data.SimResult)
```

Comment:
Should we also check that the results are correct?

Reply:
I'd say the point of these tests is more to flag simulator errors with anything that is getting netlisted by Hdl21's `Sim` class. Rather than to test that things physically make sense, although there are a couple of tests which do check that the MOSFETs are producing sensible units - so I'd say this is good enough, since scaling issues and invoking the wrong subcircuit/model is most of what can go wrong in Hdl21 PDK modules.